### PR TITLE
SG-33703 Rename user-facing authentication methods

### PR DESCRIPTION
**Pull request by julien-lang**

### CODE REVIEW

File: python/tank/authentication/login_dialog.py [hunk 1]
```diff
@@ -240,21 +240,21 @@ def __init__(
...
+            "Authenticate with the ShotGrid browser",
             menu,
         )
         self.menu_action_ulf.triggered.connect(self._menu_activated_action_web_legacy)
         menu.addAction(self.menu_action_ulf)
         menu.addAction(self.menu_action_ulf2)
 
         self.menu_action_legacy = QtGui.QAction(
-            "Authenticate with your login credentials",
+            "Authenticate with the classic credentials form",
```
Comment (shaynacohen) [inline python/tank/authentication/login_dialog.py:257]: Comapring against some notes... should this be:
```suggestion
            "Authenticate with your legacy login credentials",
```

File: python/tank/authentication/login_dialog.py [hunk 2]
```diff
@@ -533,7 +533,7 @@ def _toggle_web(self, method_selected=None):
             self.ui.login.setVisible(False)
             self.ui.password.setVisible(False)
             self.ui.message.setText(
-                "<p>Continue to sign in using your web browser.</p>"
+                "<p>Authenticate with App Session Launcher.</p>"
```
Comment (shaynacohen) [inline python/tank/authentication/login_dialog.py:536]:
```suggestion
                "<p>Authenticate with the App Session Launcher.</p>"
```

File: python/tank/authentication/login_dialog.py [hunk 3]
```diff
@@ -959,7 +959,7 @@ def run(self):
         except AuthenticationError as err:
             self.exception = err
         except Exception:
-            logger.exception("Unknown error from unified_login_flow2")
+            logger.exception("Unknown error from App Session Launcher")
```
Comment (shaynacohen) [inline python/tank/authentication/login_dialog.py:962]:
```suggestion
            logger.exception("Unknown error from the App Session Launcher")
```

**Review by shaynacohen — APPROVED** [review]
Left some suggestions. But using "Authenticate with the ShotGrid browser" works!

**Review by geffrak — APPROVED** [review]

**Pull request merged**